### PR TITLE
chore(alpha): update fields-relationship e2e tests

### PR DESCRIPTION
**Pull request by JessRynkar**
## Description

Misc fixes to the `fields-relationship` e2e test suite:

- corrects the order of props passed to `withMergedProps` function 
- adds `filterOptions` to `mergeServerFormState`
- minor changes to e2e function `runFilterOptionsTest` 

- [X] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [X] Chore (non-breaking change which does not add functionality)
- [X] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [X] Existing test suite passes locally with my changes

### CODE REVIEW

File: test/fields-relationship/config.ts [hunk 1]
```diff
@@ -193,7 +193,7 @@ export default buildConfigWithDefaults({
     },
     {
       admin: {
-        useAsTitle: 'meta.title',
+        useAsTitle: 'name',
```
Comment (jacobsfletch) [inline test/fields-relationship/config.ts:196]: Was this `useAsTitle` ever working before? When it was `meta.title`

File: packages/ui/src/elements/withMergedProps/index.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ export function withMergedProps<ToMergeIntoProps, CompleteReturnProps>({
 }): React.FC<CompleteReturnProps> {
   // A wrapper around the args.Component to inject the args.toMergeArgs as props, which are merged with the passed props
   const MergedPropsComponent: React.FC<CompleteReturnProps> = (passedProps) => {
-    const mergedProps = deepMerge(toMergeIntoProps, passedProps)
+    const mergedProps = deepMerge(passedProps, toMergeIntoProps)
```
Comment (jacobsfletch) [inline packages/ui/src/elements/withMergedProps/index.tsx:31]: Why the need to swap the order here?

File: test/fields-relationship/config.ts (hunk 1 above)
Comment (jacobsfletch) [reply test/fields-relationship/config.ts:196]: Just verified that it was not working before. Good catch.

File: packages/ui/src/elements/withMergedProps/index.tsx (hunk 2 above)
Comment (JessChowdhury) [reply packages/ui/src/elements/withMergedProps/index.tsx:31]: When this line was `deepMerge(toMergeIntoProps, passedProps)` and you check the merged output, the `path` value was getting overwritten as an empty string. When you swap the order, the merged props look correct. I can take another look however if this is not the right approach!

Comment (jacobsfletch) [reply packages/ui/src/elements/withMergedProps/index.tsx:31]: I suspect that there's a better fix for this. Where is `path` being used?

Comment (JessChowdhury) [reply packages/ui/src/elements/withMergedProps/index.tsx:31]: `path` is being used in the custom component that we are also passing to `withMergedProps`, in the component we're using `path` to set the relationship field value. Might be easier to pair and show you this.

**Pull request merged**